packages/subgraph/schema.graphql

```diff
@@ -129,3 +129,118 @@ interface ResolverEvent {
   transactionID: Bytes!
 }
 
+type AddrChanged implements ResolverEvent @entity {
+  id: ID!
+  resolver: Resolver!
+  blockNumber: Int!
+  transactionID: Bytes!
+  addr: Account!
+}
+
+type MulticoinAddrChanged implements ResolverEvent @entity {
+  id: ID!
+  resolver: Resolver!
+  blockNumber: Int!
+  transactionID: Bytes!
+  coinType: BigInt!
+  addr: Bytes!
+}
+
+type NameChanged implements ResolverEvent @entity {
+  id: ID!
+  resolver: Resolver!
+  blockNumber: Int!
+  transactionID: Bytes!
+  name: String!
+}
+
+type AbiChanged implements ResolverEvent @entity {
+  id: ID!
+  resolver: Resolver!
+  blockNumber: Int!
+  transactionID: Bytes!
+  contentType: BigInt!
+}
+
+type PubkeyChanged implements ResolverEvent @entity {
+  id: ID!
+  resolver: Resolver!
+  blockNumber: Int!
+  transactionID: Bytes!
+  x: Bytes!
+  y: Bytes!
+}
+
+type TextChanged implements ResolverEvent @entity {
+  id: ID!
+  resolver: Resolver!
+  blockNumber: Int!
+  transactionID: Bytes!
+  key: String!
+}
+
+type ContenthashChanged implements ResolverEvent @entity {
+  id: ID!
+  resolver: Resolver!
+  blockNumber: Int!
+  transactionID: Bytes!
+  hash: Bytes!
+}
+
+type DNSRecordChanged implements ResolverEvent @entity {
+  id: ID!
+  resolver: Resolver!
+  blockNumber: Int!
+  transactionID: Bytes!
+  node: Bytes!
+  name: Bytes!
+  resource: Int!
+  record: Bytes!
+}
+
+type DNSRecordDeleted implements ResolverEvent @entity {
+  id: ID!
+  resolver: Resolver!
+  blockNumber: Int!
+  transactionID: Bytes!
+  node: Bytes!
+  name: Bytes!
+  resource: Int!
+}
+
+type DNSZoneCleared implements ResolverEvent @entity {
+  id: ID!
+  resolver: Resolver!
+  blockNumber: Int!
+  transactionID: Bytes!
+  node: Bytes!
+}
+
+type DNSZonehashChanged implements ResolverEvent @entity {
+  id: ID!
+  resolver: Resolver!
+  blockNumber: Int!
+  transactionID: Bytes!
+  node: Bytes!
+  lastzonehash: Bytes!
+  zonehash: Bytes!
+}
+
+type InterfaceChanged implements ResolverEvent @entity {
```

Comment:
What does this do on resolvers?

Reply:
Oh the resolver types are stolen from ENS: https://github.com/ensdomains/ens-subgraph/blob/master/schema.graphql and are used by `resolver.ts` which already existed before.

The 3 DNS event types (DNS*) are new and not from `ens-subgraph`.

Reply:
Or was your Q more general about ResolverEvents? These are similar to DomainEvents and store changes made on different kinds of records (addr, contenthash, dns, etc.)

Reply:
I think that's related to `InterfaceResolver` that `PublicResolver` inherits from: https://github.com/ensdomains/resolvers/blob/master/contracts/profiles/InterfaceResolver.sol

It's from EIP-1844 but I don't think it's used? https://eips.ethereum.org/EIPS/eip-1844